### ISSUE TITLE
Email bodies are blank

### ISSUE THREAD

**Issue by MatthewRDodds** [britton-jb/sentinel#22]
Trying to use sentinel for invitations, I've been trying for quite awhile to get this working. At first the emails weren't sending at all, but I figured out the `send_emails` param and after setting that emails are sending but the email body is blank for both html and text formats.

This is my config:

``` elixir
config :guardian, Guardian,
  issuer: "MyApp",
  ttl: { 30, :days },
  verify_issuer: true,
  secret_key: System.get_env("SECRET_KEY"),
  serializer: MyApp.GuardianSerializer

config :guardian_db, GuardianDb,
  repo: MyApp.Repo

config :sentinel,
  app_name: "MyApp",
  user_model: MyApp.User,
  email_sender: "support@my_app.com",
  crypto_provider: Comeonin.Bcrypt,
  repo: MyApp.Repo,
  invitable: :true,
  endpoint: MyApp.Endpoint,
  router: MyApp.Router,
  send_emails: true,
  environment: :development

config :sentinel, Sentinel.Mailer,
  adapter: Bamboo.LocalAdapter

config :my_app, MyApp.Mailer,
  adapter: Bamboo.LocalAdapter
```

This is how I'm triggering it:

``` elixir
Sentinel.UserRegistration.register(%{ "user" => %{ "email" => email } })
```

**Comment by britton-jb** [comment]
Thanks for sending this my way. I'll see if I can take a look at it in the next couple days.

Depending on if you need API support or not you may also want to look at [Coherence](https://github.com/smpallen99/coherence).

**Comment by MatthewRDodds** [comment]
Thanks! I was hoping it was just a configuration problem. I might go back to where the email is converted to html/text and see whether that's where it fails, will update here with what I find.

**Comment by MatthewRDodds** [comment]
I used iex to pry inside of `confirmable_and_invitable` and set up the context to render the template, this is what that looked like:

``` elixir
email = Mailer.send_invite_email(updated_user, {confirmation_token, password_reset_token})
template = email.private.view_template
assigns = email.assigns
IEx.pry
```

I noticed in the [source for bamboo](https://github.com/thoughtbot/bamboo/blob/master/lib/bamboo/phoenix.ex#L271-L275) that the template is rendered this way `Phoenix.View.render_to_string(email.private.view_module,template,assigns)`

So I did that used that to try the render, which gave the this stack-trace:

```
** (Protocol.UndefinedError) protocol Phoenix.Param not implemented for []
     (phoenix) lib/phoenix/param.ex:125: Phoenix.Param.Any.to_param/1
    (myapp) web/router.ex:1: MyApp.Router.Helpers.user_path/4
    (myapp) web/router.ex:1: MyApp.Router.Helpers.user_url/4
    (sentinel) lib/sentinel/templates/email/invite.html.eex:6: Sentinel.EmailView."invite.html"/1
     (phoenix) lib/phoenix/view.ex:335: Phoenix.View.render_to_iodata/3
     (phoenix) lib/phoenix/view.ex:342: Phoenix.View.render_to_string/3
      (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
         (iex) lib/iex/evaluator.ex:135: IEx.Evaluator.handle_eval/6
         (iex) lib/iex/evaluator.ex:128: IEx.Evaluator.do_eval/4
         (iex) lib/iex/evaluator.ex:108: IEx.Evaluator.eval/4
         (iex) lib/iex/evaluator.ex:27: IEx.Evaluator.loop/3
         (iex) lib/iex/evaluator.ex:18: IEx.Evaluator.init/3
         (iex) lib/iex.ex:439: IEx.pry/3
```

After looking in the `invite.html` template I copied this line into iex which resulted in the same stacktrace, so that seems to be the source of my problem:

``` elixir
Sentinel.RouterHelper.helpers.user_url(Sentinel.RouterHelper.endpoint, :invited)
```

**Comment by MatthewRDodds** [comment]
I was able to solve of that error by changing my invited path in my router from `get "/users/:id/invited", UserController, :invited` to `get "/users/invited", UserController, :invited`. Now the template renders, but still without any invite link:

```
pry(1)> Phoenix.View.render_to_string(email.private.view_module,template,assigns)
"<p>Hello matthewrusselldodds@gmail.com!</p>\n\n<p>You've been invited to MyApp</p>\n"
```

**Comment by britton-jb** [comment]
@MatthewRDodds sorry about the delay, I haven't forgotten about this. My schedule has finally permitted me to take a look. I'm able to replicate the issue, and I'm working on a fix for it. Unfortunately as you can likely guess I don't have a timeframe on it just yet.

**Comment by MatthewRDodds** [comment]
No worries, thanks for the update @britton-jb! I ended up going ahead and writing my own app specific invite routine. Maybe it was something to do with my local configuration or implementation, in that case it might be good to close this for now and see if other people have trouble.

**Comment by britton-jb** [comment]
Unfortunately this is definitely a bug that I need to fix and include a spec for, based on my testing. Thanks for letting me know, and I'm glad you were able to find a workaround in the meantime.

**Comment by britton-jb** [comment]
Fixed in the v2 branch, which should have a 2.0.0 release in the coming week.
